pkg/workflow/workflow.go

```diff
@@ -451,26 +500,28 @@ func (e *engine) steps(taskRunners []wfTypes.TaskRunner) error {
 
 		e.failedAfterRetries = e.failedAfterRetries || operation.FailedAfterRetries
 		e.waiting = e.waiting || operation.Waiting
-		if status.Phase != common.WorkflowStepPhaseSucceeded {
-			wfCtx.IncreaseCountValueInMemory(wfTypes.ContextPrefixBackoffTimes, status.ID)
+		if status.Phase == common.WorkflowStepPhaseSucceeded || (status.Phase == common.WorkflowStepPhaseRunning && status.Type == wfTypes.WorkflowStepTypeSuspend) {
+			wfCtx.DeleteValueInMemory(wfTypes.ContextPrefixBackoffTimes, status.ID)
```

Comment:
Why we gonna delete the backoff times if the step is delay suspend?

Reply:
the delay suspend step will never run(task runner exec run)again,  we should delete backoff times after running.

Reply:
In workflow ExecuteSteps, it will return common.WorkflowStateSuspended before exec run when wfStatus.Suspend is true.

Reply:
I think it's better to wrap `(status.Phase == common.WorkflowStepPhaseRunning && status.Type == wfTypes.WorkflowStepTypeSuspend)` to a function like `isWaitSuspendStep()`, otherwise it might be confusing to other developers.

Reply:
it's right, i'll fix it